kratos/includes/table_accessor.h

```diff
@@ -0,0 +1,205 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Alejandro Cornejo
+//                   Riccardo Rossi
+//                   Carlos Roig
+//                   Ruben Zorrilla
+//
+
+# pragma once
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/accessor.h"
+
+namespace Kratos
+{
+///@addtogroup KratosCore
+///@{
+
+///@name Kratos Classes
+///@{
+
+/**
+ * @class TableAccessor
+ * @ingroup Kratos Core
+ * @brief This class defines the way a certain property is accessed according to a table.
+ * @brief The tables are supposed to relate double <-> double type entities. 
+ * @brief The input variable is suposed to be a nodally accesible one (either historical or not)
+ * @author Alejandro Cornejo, Riccardo Rossi, Carlos Roig and Ruben Zorrilla
+ */
+class KRATOS_API(KRATOS_CORE) TableAccessor : public Accessor
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Geometry type definition
+    using GeometryType = Geometry<Node>;
+
+    /// Variable type
+    using VariableType = Variable<double>;
+
+    /// BaseType
+    using BaseType = Accessor;
+
+    using SizeType = std::size_t;
+
+    enum class InputVariableType
+    {
+        NodalHistorical = 0,
+        NodalNonHistorical = 1,
+        ElementalNonHistorical = 2
+    };
+
+    /// Pointer definition of TableAccessor
+    KRATOS_CLASS_POINTER_DEFINITION(TableAccessor);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor
+    TableAccessor() = default;
+
+    /// Custom constructor
+    TableAccessor(const VariableType& rInputVariable, const std::string& rInputVariableType = "nodal_historical") 
+        : mpInputVariable(std::make_shared<VariableType>(rInputVariable)) 
```

Comment:
Why shared?

Reply:
definitely it should be either a row pointer or a const reference